src/Aspire.Hosting/Dcp/DcpExecutor.cs

```diff
@@ -1048,12 +1048,12 @@ private void PrepareProjectExecutables()
                 {
                     exeSpec.Spec.ExecutionType = ExecutionType.Process;
 
-                    if (_configuration.GetBool("DOTNET_WATCH") is not true)
+                    if (_configuration.GetBool("DOTNET_WATCH") is not true || projectMetadata.IsFileBasedApp)
                     {
                         projectArgs.AddRange([
                             "run",
                             "--no-build",
```

Comment:
Need to change this so that `--no-build` isn't passed if we're doing file-based apps. We might want to revisit it for all projects too so that `dotnet run` can do the build rather than us relying on project references from the AppHost project.